mininet/node.py

```diff
@@ -839,6 +839,48 @@ def init( cls ):
         cls.inited = True
 
 
+class XIAHost( Host ):
+    "eXpressive Internet Architecture host"
+
+    def __init__( self, name, **kwargs ):
+        Node.__init__( self, name, **kwargs )
+        self.xids = {}  # mapping of configurable inputs to XID types
+
+    def config( self, hid=None, xdp=False, **params):
+        """hid: private key file names for Host Identifiers, eg. hid='pk1,pk2'
+           xdp: set True to load the xdp module
+           params: parameters for Node.config()"""
+        r = Node.config( self, **params )
+        if xdp is True:
+            moduleDeps( add='xia_ppal_xdp')
+        self.setParam( r, 'setHID', hid=hid )
+        return r
+
+    def setHID( self, hid=None ):
+        "Assign HID(s) to the node."
+        self.xids[ 'hid' ] = self.addParameters( hid )
+        # Load the required module
+        moduleDeps( add='xia_ppal_hid' )
+        # Performs actual HID configuration
+        for p in self.xids[ 'hid' ]:
+            self.cmd( 'xip hid new', p )
+            self.cmd( 'xip hid addaddr', p )
+
+    def addParameters( self, xids ):
+        """Store the ',' separated xids as a list.
+           xids: string input for XID(s)"""
+        xidlist = []
+        xids = xids.split( ',' )
+        for xid in xids:
+            xidlist.append( xid )
+        return xidlist
+
+    def cleanup( self ):
+        "Unload the modules loaded by the principals."
+        moduleDeps( subtract='xia_ppal_hid' )
```

Comment:
no checks made while unloading the modules.  moduleDeps() handles unloading a module that is absent quite gracefully.